src/CrawlInternalWithSubdomainUrls.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Spatie\Crawler;
+
+class CrawlInternalWithSubdomainUrls implements CrawlProfile
+{
+    protected $host = '';
+
+    public function __construct(string $baseUrl)
+    {
+        $this->host = parse_url($baseUrl, PHP_URL_HOST);
+    }
+
+    public function shouldCrawl(Url $url): bool
+    {
+        return substr($url->host, -strlen($this->host)) === $this->host;
```

Comment:
Maybe extract this line into its own method to improve readability

Reply:
this method can then be tested instead of the `profile_crawls_a_domain_and_its_subdomains`